text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+- **The constructor is `undefined` or `null`**
+    - When the constructor is undefined the `<lwc.dynamic>` tag along with all of its children are not rendered to the DOM.
+- **The constructor is defined and is not a `LightningElementConstructor`**
+    - When the value provided to `lwc:is` is not a `LightningElementConstructor` then an error is thrown indicating the constructor value must be `LightningElementConstructor`.
+- **The constructor is defined and is a `LightningElementConstructor`**
+    - When the constructor is a `LightningElementConstructor`, the component will be rendered in place of `<lwc.dynamic>`. The tag name used for the dynamic component is the value of `lightningElementName` on `LightningElementConstructor`.
+
+_Note in the case of lazy loading, the component author is responsible for resolving the promise.  The only value that can be given to `lwc:is` is a `LightningElementConstructor`._
+
+#### Selecting the dynamic component
+
+It is important to note that `<lwc.dynamic>` will not be rendered to the DOM.  As such, it is not available to be selected using `this.template.querySelector`.  The actual custom element must be selected once it has been rendered to the DOM.  The dynamic component can be selected either through the `lightningElementName` property on the constructor or through another attribute assigned to the component.
+
+Component authors can use the `connectedCallback` of the dynamic component to signal when the component has been rendered to the DOM.
+
+#### Assigning attributes and event listeners
+
+Attributes and event listeners can be passed to the `<lwc.dynamic>` tag declaratively or can be assigned directly to the custom element imperatively.
+
+Attributes and event listeners assigned declaratively will be assigned to the dynamic component once it has been created.
+
+Declaratively:
+
+```html
+<template>
+    <lwc.dynamic lwc:is={ctor} class="container" onclick={handleClick}></lwc.dynamic>
+</template>
+```
+
+Imperatively:
+
+```javascript
+import { LightningElement } from "lwc";
+
+export default class DynamicCtor extends LightningElement {
+  customCtor;
+
+  connectedCallback() {
+    this.loadCtor();
+  }
+
+  async loadCtor() {
+    const { default: ctor } = await import("c/customConstructor");
+    this.customCtor = ctor;
+    this.template.querySelector(ctor.lightningElementName).setAttribute('title', 'hawaii');
+    this.template.querySelector(ctor.lightningElementName).addEventListener('click', () => console.log('hello world'));
+  }
+}
+```
+#### Usage with other LWC directives
+
+All [LWC custom element directives](https://lwc.dev/guide/reference#directives-for-nested-templates) with the exception of `lwc:external` are available to use on `<lwc.dynamic>` and their functionality will be passed through to the custom element once it has been created.
```

Comment:
I'm a bit unclear what directives are or are not included here. For instance, is `lwc:spread` included?

Reply:
I don't see a reference to `lwc:ref` in the lwc.dev site and I wanted to reference the guide to ensure that I covered all the directives.  

I'll check to see if there's another source with a comprehensive list, otherwise, I can list them out here.

Reply:
I'll add it to the docs. It should be there.

Reply:
@nolanlawson updated this section, let me know if it's clearer now which directives are supported.